src/plugins/plugin.filler.js

```diff
@@ -190,37 +190,74 @@ function isDrawable(point) {
 	return point && !point.skip;
 }
 
-function drawArea(ctx, curve0, curve1, len0, len1) {
+function drawArea(ctx, curve0, curve1, len0, len1, mode, area) {
 	var i;
+	var fillAreaPointsSet = [];
+	var clipAreaPointsSet = [];
+	var above = mode === 'above';
+	var below = mode === 'below';
 
 	if (!len0 || !len1) {
 		return;
 	}
 
+	if (above) {
+		clipAreaPointsSet.push({x: curve1[len1 - 1].x, y: area.top});
+	} else if (below) {
+		clipAreaPointsSet.push({x: curve0[0].x, y: area.top});
+		clipAreaPointsSet.push(curve0[0]);
+	}
+
 	// building first area curve (normal)
-	ctx.moveTo(curve0[0].x, curve0[0].y);
+	fillAreaPointsSet.push(curve0[0]);
 	for (i = 1; i < len0; ++i) {
-		helpers.canvas.lineTo(ctx, curve0[i - 1], curve0[i]);
+		curve0[i].flip = false;
+		fillAreaPointsSet.push(curve0[i]);
+		if (below) {
+			clipAreaPointsSet.push(curve0[i]);
+		}
 	}
 
 	// joining the two area curves
-	ctx.lineTo(curve1[len1 - 1].x, curve1[len1 - 1].y);
+	curve1[len1 - 1].joint = true;
+	fillAreaPointsSet.push(curve1[len1 - 1]);
 
 	// building opposite area curve (reverse)
 	for (i = len1 - 1; i > 0; --i) {
-		helpers.canvas.lineTo(ctx, curve1[i], curve1[i - 1], true);
+		if (above) {
+			curve1[i].flip = true;
+			clipAreaPointsSet.push(curve1[i]);
+		}
+		curve1[i - 1].flip = true;
+		fillAreaPointsSet.push(curve1[i - 1]);
 	}
+	if (above || below) {
```

Comment:
Same `if(above) {..} else if(below) {..}` here to reduce nesting (and complexity)